### PR TITLE
Add Guilded v1.0.9104602-release

### PR DESCRIPTION
**Pull request by abheekda1**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by ADawesomeguy** [comment]
This is my first time contributing a cask and one of my first times contributing to an open-source project other than one of my own. Please let me know if I did something wrong!

**Comment by bevanjkay** [comment]
Thanks @ADawesomeguy. It's a good start!

Can you please try to add a [`livecheck` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-livecheck) and a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap)

**Comment by ADawesomeguy** [comment]
Thanks! I added a basic `zap` stanza. However, I'm a little confused about the `livecheck` stanza. Is that needed due to the fact that the Guilded download does not have a versioned URL?

**Comment by bevanjkay** [comment]
@ADawesomeguy the zap stanza looks great. We use livecheck to compare the version in the cask with the version from upstream. This ensures that version updates are accurate (CI runs the comparison and fails if there is a difference), and also allows the ability to easily check for newer versions with the `brew livecheck` command. 

I can have a look at adding a livecheck stanza for you shortly.

File: Casks/guilded.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+cask "guilded" do
+  version "1.0.9104602-release"
```
Comment (SMillerDev) [inline Casks/guilded.rb:2]: I don't think `-release` should be part of the version?

Comment (bevanjkay) [reply Casks/guilded.rb]: @SMillerDev I have updated this now. The :electron_builder strategy for livecheck was sometimes returning the version with the `-release` and sometimes without. But it seems to be consistently returning it without now.

Comment (SMillerDev) [reply Casks/guilded.rb]: Yeah, sometimes upstream does the weirdest stuff. That's why I asked, thanks for the fix.

Comment (bevanjkay) [reply Casks/guilded.rb]: @SMillerDev If the current iteration doesn't work I have another one with a `do` block on the strategy to strip the `-release`. It was returning without `-release` on `brew livecheck` but with `-release` for `brew audit` 🤷🏻‍♂️

**Review by SMillerDev — APPROVED** [review]
LGTM, leaving this open for other maintainers though

**Pull request merged**